_fr/tour/abstract-type-members.md

```diff
@@ -2,11 +2,80 @@
 layout: tour
 title: Abstract Type Members
 partof: scala-tour
-
-num: 21
-
+num: 25
 language: fr
-
 next-page: compound-types
 previous-page: inner-classes
+topics: abstract type members
+prerequisite-knowledge: variance, upper-type-bound
+
+redirect_from: 
+  - "/tutorials/tour/abstract-types.html"
+  - "/tour/abstract-types.html"
+
 ---
+
+Les types abstraits, tels que les traits et les classes abstraites, peuvent avoir des membres de type abstrait.
+Cela signifie que les implémentations concrètes définissent les types réels.
+Voici un exemple :
+
+```scala mdoc
+trait Buffer {
+  type T
+  val element: T
+}
+```
+
+Ici, nous avons défini un `type T` abstrait. Il est utilisé pour décrire le type de `element`. Nous pouvons étendre ce trait dans une classe abstraite, en ajoutant une limite de type supérieure à `T` pour le rendre plus spécifique.
```

Comment:
```suggestion
Ici, nous avons défini un `type T` abstrait. Il est utilisé pour décrire le type de `element`. Nous pouvons étendre ce trait dans une classe abstraite, en ajoutant une borne de type supérieure à `T` pour le rendre plus spécifique.
```
I'm not sure. Intuition tells me "borne de type" is better than "limite de type".